src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java

```diff
@@ -298,4 +319,33 @@ private static ConfigDef createConfigDef() {
     MongoSinkTopicConfig.BASE_CONFIG.configKeys().values().forEach(configDef::define);
     return configDef;
   }
+
+  /**
+   * This method is used in our implementation of the {@link ConfigDef#validateAll(Map)} method
+   * because the original implementation silently disregards any undefined configuration properties.
+   *
+   * @param propertyNames Either normal configuration property names or property names prefixed with
+   *     {@link MongoSinkTopicConfig#TOPIC_OVERRIDE_PREFIX} and a topic name, as specified <a
+   *     href="https://docs.mongodb.com/kafka-connector/master/sink-connector/configuration-properties/topic-override/">here</a>.
+   * @see #OBSOLETE_CONFIGS
+   */
+  static void logObsoleteProperties(final Collection<String> propertyNames) {
```

Comment:
I am not sure if any configuration properties have been made obsolete up until now, and if so, what is the process. I think warning a user about still having these properties is useful, but if we have a different process of obsoleting configuration properties, please share.